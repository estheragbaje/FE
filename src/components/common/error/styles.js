import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  background-color: #fff;
  justify-content: center;
  align-items: center;
   flex-direction: column;
  button {
    width: 150px;
    border: 1px solid #63beff;
    padding: 0.5rem;
    margin: 2rem;
    font-size: 1.6rem;
    color: #63beff;
    outline: 0;
    font-weight: 500;
    background-color: transparent;
    &:hover {
      color: #fff;
      background-color:#63beff;
      transition: 0.5s ease-in-out;
    }
    span {
      padding-left: 0.8rem;
    }
  }
  div {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    img {
      width: 60%;
      height: 60%;
      object-fit: contain;
    }
  }
`;
