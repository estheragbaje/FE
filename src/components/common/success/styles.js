import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  background-color: #fff;
  justify-content: space-between;
  align-items: center;
  @media (min-width: 320px) and (max-width: 768px) {
    flex-direction: column-reverse;
    flex-wrap: nowrap;
  }
  div:nth-of-type(1) {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    flex-basis: 30%;
    width: 100%;
    height: 100%;
    p {
      line-height: 32px;
      padding-left: 4rem;
      font-size: 1.6rem;
      text-align: center;
    }
    @media (max-width: 500px) {
      flex-basis: 100%;
      p {
        padding: 3rem;
      }
    }
  }
  div:nth-of-type(2) {
    flex-basis: 70%;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    img {
      width: 60%;
      height: 60%;
      object-fit: contain;
    }
    @media (max-width: 500px) {
      flex-basis: 100%;
      img {
        width: 100%;
        height: 100%;
      }
    }
  }
`;
