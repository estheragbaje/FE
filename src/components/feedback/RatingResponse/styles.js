import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  width: 100%;
  margin: 2rem auto;
  display: flex;
  flex-direction: column;
`;
export const Label = styled.div`
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  div:first-of-type {
    color: #E02F3A;
  }
  div:nth-of-type(2) {
    color: #76A80B;
  }
`;

export const Wrapper = styled.div`
  border: 1px solid #62beff;
  width: 100%;
  display: flex;
  justify-content: space-around;
  padding: 0.5rem;
`;

export const RatingWrapper = styled.div`
  padding: 0.5rem;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid #62beff;
  transition: all 0.2s ease-in-out;
  z-index: 1000;
  &:hover {
    cursor: pointer;
  }
  &.selected {
    background: #62beff;
    color: white;
    transition: background-color 0.2s ease-in-out;
  }
  @media (max-width: 500px) {
    width: 24px;
    height: 24px;
  }
`;
