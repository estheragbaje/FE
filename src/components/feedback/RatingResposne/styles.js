import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  width: 100%;
  margin: 2rem auto;
  display: flex;
  flex-direction: column;
`;
export const Label = styled.div`
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  div:first-of-type {
    color: red;
  }
  div:nth-of-type(2) {
    color: green;
  }
`;

export const Wrapper = styled.div`
  border: 1px solid #43bccd;
  width: 100%;
  display: flex;
  justify-content: space-around;
  padding: 0.5rem;
`;

export const RatingWrapper = styled.div`
  padding: 0.5rem;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid #43bccd;
  transition: all 0.2s ease-in-out;
  z-index: 1000;
  &:hover {
    cursor: pointer;
  }
  &.selected {
    background: #43bccd;
    color: white;
    transition: background-color 0.2s ease-in-out;
  }
  @media (max-width: 500px) {
    width: 22px;
    height: 22px;
  }
`;
