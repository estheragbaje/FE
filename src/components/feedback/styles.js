import styled from 'styled-components';

export const Container = styled.div`
  width: 804px;
  background: #ffffff;
  margin: 4rem auto;
  @media (max-width: 500px) {
    width: 100%;
  }
  display: flex;
  flex-direction: column;
  padding: 68px;
`;

export const Title = styled.h2`
  font-family: Baloo Thambi;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 28px;
  text-align: center;
  position: relative;
  padding-bottom: 30px;
  &::before {
    position: absolute;
    content: '';
    background: #cbd0dc;
    bottom: 0;
    left: 50%;
    width: 129px;
    height: 1px;
    transform: translateX(-50%);
  }
`;

export const TextResponseWrapper = styled.div`
  width: 100%;
  background: #ffffff;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 2rem auto;
  justify-content: flex-start;
`;

export const Text = styled.p`
  font-family: Miriam Libre;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 24px;
  color: #000000;
  width: 100%;
`;

export const TextArea = styled.textarea`
  border: 1px solid #ccc;
  width: 100%;
  padding: 0.8rem;
`;
