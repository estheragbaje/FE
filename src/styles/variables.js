export const mainColor = '#63BEFF';