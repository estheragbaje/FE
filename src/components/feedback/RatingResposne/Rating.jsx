import React from 'react';
import propTypes from 'prop-types';

import { RatingWrapper } from './styles';

export default function Rating({ value, selectHandler }) {
  return (
    <RatingWrapper onClick={(e) => selectHandler(e)} className="rating">
      <span className="circle" />
      <span className="number">{value}</span>
    </RatingWrapper>
  );
}

Rating.propTypes = {
  value: propTypes.string.isRequired,
  selectHandler: propTypes.func.isRequired,
};
