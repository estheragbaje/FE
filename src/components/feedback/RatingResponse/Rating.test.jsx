import React from 'react';
import { shallow } from 'enzyme';
import { RatingWrapper } from './styles';
import Rating from './Rating';

describe('<Rating />', () => {
  const props = {
    value: 2,
    selectHandler: jest.fn(),
  };
  // eslint-disable-next-line react/jsx-props-no-spreading
  const wrapper = shallow(<Rating {...props} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    const children = wrapper.find(RatingWrapper).children();
    expect(children).toHaveLength(2);
    expect(children.find('span')).toHaveLength(2);
  });
});
