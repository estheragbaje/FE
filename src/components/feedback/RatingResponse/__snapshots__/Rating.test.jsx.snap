// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should render correctly 1`] = `
<styled.div
  className="rating"
  onClick={[Function]}
>
  <span
    className="circle"
  />
  <span
    className="number"
  >
    2
  </span>
</styled.div>
`;
